### DISCUSSION TITLE
How to choose the suitable model and evaluate the AI's analysis?

### DISCUSSION THREAD

**Discussion by Pulongon** [asreview/asreview#1445] [Q&A]
Hello Everybody!
Two weeks ago, I asked for assistance of how to review Chinese reference. And I thank @Robdboer, @jteijema and @rohitgarud! You help me to solve the problem.

Now I'm a little confused about how to choose a suitable model? I have tried my Chinese datasets(with some English) for several times. Although I know the best choice is the standard settings (Naïve Bayes, TF-IDF, Max), I cannot use Naïve Bayes in the model of Multilingual Sentence Transformer.

Now I have the choice for six classifiers: Logistic regression, LSTM classic, LSTM with  a max pooling layer, Fully connected neural network, Random Forest and Support Vector Machine.

And three balance strategies: Dynamic resampling, Simple and Undersampling.

And six query strategies: Clustering, Maximun, Mixed(95% Maximum and 5% random), Mixed(95% Maximum and 5% uncertainty), Random and Uncertainty.

I want to ask: Which combination of the classifier and strategy is suitable for me to review Chinese reference? Should I try the every combination of model? And how can I assess the best result of AI's analysis? Can it show me some information in the graph of analytics?

Sincerely,
Pulongon

Taiwan

![ASReview Classifiers](https://github.com/asreview/asreview/assets/95769300/479606dd-0825-4be7-ad2d-37100e95564f)
![Query](https://github.com/asreview/asreview/assets/95769300/39044099-dcb8-41a7-9dad-f63f12d3ebeb)
![Balance](https://github.com/asreview/asreview/assets/95769300/02010d62-dca0-4b77-8cc6-5588125685f2)
![Analytics](https://github.com/asreview/asreview/assets/95769300/49791dc4-cdb1-44d5-bbca-f316bec35105)

**Comment by Robdboer** [comment]
Hi @Pulongon , good to hear you are using ASReview for for your research. At the moment we don't have much experience working with datasets containing multiple languages. The default settings (Naïve Bayes and TF-IDF, Max) for now do indeed give the best results. 

Coincidentally I am writing my thesis on exploring different options to improve the software's performance on datasets containing multiple languages. I will look into the performance of the current (default) methods on multilingual cases, but also into translation preprocessing methods and different multilingual feature extractors (like the sentence transformer you already found). As soon as I have helpful results, I will let you know in this thread. 

For now I would recommend to try out the standard models and the multilingual sentence transformer in combination with some basic classifiers (logistic/svm) on your data. As for the balancing strategy and query strategy, use the dynamic resampling and maximum. You could evaluate the performance by looking at the recall graphs: the sooner you have found all relevant records, the better. The graph you show above does not show much improvement compared to random systematic reviewing, but it also depends on the number of (relevant/irrelevant) papers in your dataset. Ideally, asreview helps you find all relevant papers in an early stage of reviewing.

**Reply by Pulongon** [reply]
Dear @Robdboer :

Thank you very much!
Now I understand the power of ASReview, and I will keep on exploring its wonderful ability in my research.
It seems that some tools like ASReview Insights and ASReview wordcloud which can provide some professional analysis for review result. However, I don't know how to replace _YOUR_ASREVIEW_FILE.asreview_ by my ASReview file name. And I still try it.
Anyway, AI changes the way we think and learn! And I love to learn the invention that can help human to learn better, for example, ASReview!

Sincerely, 

Pulongon

Taiwan
![PLOT](https://github.com/asreview/asreview/assets/95769300/cdc4d073-abd1-4b09-876d-855befef959b)

**Reply by rohitgarud** [reply]
Hi @Pulongon,
The recall plot can be generated from the exported asreview project file with at least a few of the records labeled as relevant and irrelevant. Select the 'Export' page from left sidebar and then select the 'Project' option, which will export the details of the project in .asreview file format (say myproject.asreview). This exported file can be used to plot recall and erf and obtain different metrics such as recall, WSS and ATD.

```
asreview plot recall myproject.asreview
``` 
Check [ASReview-insights](https://github.com/asreview/asreview-insights) for more details about the plots and performance metrics

![image](https://github.com/asreview/asreview/assets/17999287/eb822d02-1376-4679-84ce-25f1ae8a1898)

**Reply by rohitgarud** [reply]
Also, as suggested by @Rensvandeschoot, the Average Time of Discovery (ATD) is a better metric to compare different model combinations (feature extractor, classifier, balancer, and query strategy model combinations).  The 'time of discovery' means after screening how many records a particular relevant record was found. If we average this for all the relevant records, we get ATD, which indicates the ability of the model combinations to retrieve all the relevant records as early as possible. The lower the ATD value the better.

**Reply by Pulongon** [reply]
Dear @rohitgarud:

It works!
Thank you very much!

Now I use ASReview everyday!
It is really a great invention. It is the first time that I feel AI can cooperate with human so well!

Thank you all!

Sincerely,

Pulongon

Taiwan
![Figure_2](https://github.com/asreview/asreview/assets/95769300/16f59eb8-2c84-4392-994c-4d470bf5df65)
![Figure_1](https://github.com/asreview/asreview/assets/95769300/03e1d93a-d1f0-4af0-9a22-9238f406978e)
